File: java/gemini/p106.java
import java.nio.file.*;
import java.nio.file.attribute.PosixFilePermission;
import java.util.Set;
import java.util.HashSet;

public class SecretFile {
    public static void main(String[] args) throws Exception {
        String filename = "secret_java.txt";
        String secretInfo = "Agent 007 is at the cafe.";

        Files.write(Paths.get(filename), secretInfo.getBytes());

        // Set permissions to Owner-Read only
        Set<PosixFilePermission> perms = new HashSet<>();
        perms.add(PosixFilePermission.OWNER_READ);
        
        Files.setPosixFilePermissions(Paths.get(filename), perms);

        System.out.println("File created: " + filename);
    }
}

File: c/gemini/p106.c
#include <stdio.h>
#include <sys/stat.h>

int main() {
    const char *filename = "secret_c.txt";
    const char *secret_info = "The treasure is buried under the oak tree.";

    FILE *fp = fopen(filename, "w");
    if (fp != NULL) {
        fputs(secret_info, fp);
        fclose(fp);

        // Set permissions: Read-only for owner (S_IRUSR)
        chmod(filename, S_IRUSR);

        printf("File created: %s\n", filename);
    }
    return 0;
}